src/sentry/web/urls.py

```diff
@@ -536,8 +536,8 @@
                     RestoreOrganizationView.as_view(),
                     name="sentry-restore-organization",
                 ),
-                # need to catch settings and force it to react
-                url(r"^(?P<organization_slug>[\w_-]+)/settings/", react_page_view),
+                # need to catch `settings`, `discover`, etc., and force them to react
+                url(r"^(?P<organization_slug>[\w_-]+)/[\w_-]+/", react_page_view),
```

Comment:
This feels pretty broad. Is it too broad?

Reply:
I think it being broad is what helps get SSO users to the right place when they don't have an active session though.

Reply:
Yes.

If this regex catches some nonsense address that it wouldn't have previously, logged-in users will still see the 404 page from `react_page_view`. Logged-out users would be taken to a login page either way (and would be redirected back to the 404 view after validating); this only affects whether they see the org SSO page.

Making it more specific (such as `r"^(?P<organization_slug>[\w_-]+)/(settings|discover)/"`) would, AFIACT, only have the effect of causing SSO users to see their org's login page if they land on an address that exists and the non-SSO login page if they land on one that doesn't.

Reply:
Having a more specific pattern is going to be a long term maintenance headache as well.